### ISSUE TITLE
Menu (Uncaught Error: React DOM tree root should always have a node reference.)

### ISSUE THREAD

**Issue by fendyjong** [grommet/grommet#1611]
Menu is giving an error in the console. After clicking the menu item with path

```
<Menu icon={<AddIcon />} dropAlign={{ "right": "right" }}>
    {quickAddMenu.map(menu => (
        <Anchor key={shortid.generate()} path={menu.path}>{menu.label}</Anchor>
    ))}
</Menu>
```

### Actual Behavior
error in the console:
> Uncaught Error: React DOM tree root should always have a node reference.
    at invariant (index.js:987)
    at Object.getNodeFromInstance (index.js:2544)
    at Object.didPutListener (index.js:65266)
    at Object.putListener (index.js:12220)
    at Object.putListener (index.js:59100)
    at CallbackQueue.notifyAll (index.js:33140)
    at ReactReconcileTransaction.close (index.js:64053)
    at ReactReconcileTransaction.closeAll (index.js:15811)
    at ReactReconcileTransaction.perform (index.js:15758)
    at ReactUpdatesFlushTransaction.perform (index.js:15745)
    at ReactUpdatesFlushTransaction.perform (index.js:6237)
    at Object.flushBatchedUpdates (index.js:6320)
    at ReactDefaultBatchingStrategyTransaction.closeAll (index.js:15811)
    at ReactDefaultBatchingStrategyTransaction.perform (index.js:15758)
    at Object.batchedUpdates (index.js:63719)
    at Object.batchedUpdates (index.js:6245)
    at dispatchEvent (index.js:63877)

### Steps to Reproduce
 1. Click on one of the menu item

### Your Environment

* Grommet version: 1.3.4
* Browser Name and version: Chrome Version 60.0.3112.113
* Operating System and version (desktop or mobile): Mac Version 10.12.6
* React Version: 15.4.0
* React Router Version: 4.2.0

**Comment by karatechops** [comment]
Thanks for filing this issue, @fendyjong. I can confirm this is happening in my local grommet docs setup. It does not appear to be be happening in a Grommet project I'm developing in Grommet 1.6.0 so that may provide a clue as to where the bug originated. I'll do a bit more digging and hopefully post a fix soon.

**Comment by RyanCCollins** [comment]
Okay if I close this?

**Comment by karatechops** [comment]
@RyanCCollins let's wait for the PR merge. It should auto close as its tagged in the commit message.

**Comment by fendyjong** [comment]
@karatechops it's supposedly grommet version 1.7.0
I was wrong. I was taking the version from the json.
I just recheck in the npm version and it's grommet version 1.7.0

**Comment by karatechops** [comment]
@fendyjong, thanks for the info. I've posted a PR with a fix for this (#1611), it will be available in our [latest stable](https://github.com/grommet/grommet/wiki/What-is-grommet-stable-and-how-to-use-it%3F) after the PR merges.

I was not able to nail this down to a specific piece of code which was introduced in 1.7.0. I assume a package inadvertently updated which created a race condition in React regarding handling the click event and the Menu DOM element being removed.

**Comment by RyanCCollins** [comment]
@karatechops It's like watching grass grow 😄  (waiting for the bot to close this, that is).
